tasks/coverage/snapshots/parser_test262.snap

```diff
@@ -21017,13 +21024,27 @@ Expect Syntax Error: tasks/coverage/test262/test/language/import/import-attribut
   × Unexpected flag G in regular expression literal
     ╭─[test262/test/language/literals/regexp/early-err-bad-flag.js:19:4]
  18 │ 
+ 19 │ /./G;
+    ·    ─
+    ╰────
+
```

Comment:
Because new regexp parser also validates flags, it seems invalid flags are reported twice. 

Should I do something?

Reply:
Interesting. We need to turn this error off in `oxc_regular_expression` if it is called from the parser.

Reply:
Between those two, the latter feels more natural to me.

But I'd like to avoid depending error message strings to check like this...

```diff
         {
             Ok(regular_expression) => Some(self.ast.alloc(regular_expression)),
             Err(diagnostic) => {
-                self.error(diagnostic);
+                if !diagnostic.message.contains("flags") {
+                    self.error(diagnostic);
+                }
                 None
             }
         }
```

Do you have any ideas?

What about

- add the flag that the lexer detected an invalid regexp flags
- propagate it and then not parsing the regular expression in the first place

```diff
     pub(crate) fn parse_literal_regexp(&mut self) -> Result<RegExpLiteral<'a>> {
         let span = self.start_span();
         // split out pattern
-        let (pattern_end, flags) = self.read_regex()?;
+        let (pattern_end, flags, flags_invalid) = self.read_regex()?;
         let pattern_start = self.cur_token().start + 1; // +1 to exclude left `/`
         let pattern_text = &self.source_text[pattern_start as usize..pattern_end as usize];
         let flags_start = pattern_end + 1; // +1 to include right `/`
         let flags_text = &self.source_text[flags_start as usize..self.cur_token().end as usize];
         self.bump_any();
-        let pattern = self
-            .options
-            .parse_regular_expression
+        let pattern = (self.options.parse_regular_expression && !flags_invalid)
             .then_some(())
             .map(|()| {
                 self.parse_regex_pattern(pattern_start, pattern_text, flags_start, flags_text)

```

Reply:
* add the flag that the lexer detected an invalid regexp flags
* propagate it and then not parsing the regular expression in the first place

let's try this

Reply:
👉🏻  843d6af5269c4dd0eb3087ef3686e51879c4c5b3